config/PropertyNames.xml

```diff
@@ -195,363 +195,364 @@ generated from the section label.
+        <suffix name="InvocationTimeout" languages="all" />
+        <suffix name="Locator" languages="all" />
+        <suffix name="Router" languages="all"/>
+        <!-- TODO: Remove from javascript -->
+        <suffix name="CollocationOptimized" languages="all" />
+        <suffix name="Context.[any]" languages="all" />
     </class>
 
     <class name="connection" prefix-only="true">
-        <suffix name="CloseTimeout" default="10" />
-        <suffix name="ConnectTimeout" default="10" />
-        <suffix name="EnableIdleCheck" default="1" />
-        <suffix name="IdleTimeout" default="60" />
-        <suffix name="InactivityTimeout" default="300" />
-        <suffix name="MaxDispatches" default="100" />
+        <suffix name="CloseTimeout" languages="all" default="10" />
+        <suffix name="ConnectTimeout" languages="all" default="10" />
+        <suffix name="EnableIdleCheck" languages="all" default="1" />
+        <suffix name="IdleTimeout" languages="all" default="60" />
+        <suffix name="InactivityTimeout" languages="all" default="300" />
+        <suffix name="MaxDispatches" languages="cpp,csharp,java" default="100" />
     </class>
 
     <class name="threadpool" prefix-only="true">
-        <suffix name="Size" default="1" />
-        <suffix name="SizeMax" />
-        <suffix name="SizeWarn" default="0" />
-        <suffix name="StackSize" default="0" />
-        <suffix name="Serialize" default="0" />
-        <suffix name="ThreadIdleTime" default="60" />
-        <suffix name="ThreadPriority" />
+        <suffix name="Size" languages="cpp,csharp,java" default="1" />
+        <suffix name="SizeMax" languages="cpp,csharp,java" />
+        <suffix name="SizeWarn" languages="cpp,csharp,java" default="0" />
+        <!-- TODO: remove from c++ -->
+        <suffix name="StackSize" languages="csharp,java" default="0" />
+        <suffix name="Serialize" languages="cpp,csharp,java" default="0" />
+        <suffix name="ThreadIdleTime" languages="cpp,csharp,java" default="60" />
+        <!-- TODO: remove from c++ -->
+        <suffix name="ThreadPriority" languages="csharp,java" />
     </class>
 
     <class name="objectadapter" prefix-only="true">
-        <suffix name="AdapterId" />
-        <suffix name="Connection" class="connection"/>
-        <suffix name="Endpoints" />
-        <suffix name="Locator" class="proxy"/>
-        <suffix name="PublishedEndpoints" />
-        <suffix name="ReplicaGroupId" />
-        <suffix name="Router" class="proxy"/>
-        <suffix name="ProxyOptions" />
-        <suffix name="ThreadPool" class="threadpool" />
-        <suffix name="MessageSizeMax" />
+        <suffix name="AdapterId" languages="cpp,csharp,java" />
+        <suffix name="Connection" class="connection" languages="cpp,csharp,java"/>
+        <suffix name="Endpoints" languages="all" />
+        <suffix name="Locator" class="proxy" languages="cpp,csharp,java" />
+        <suffix name="PublishedEndpoints" languages="all" />
```

Comment:
PublishedEndpoints is indeed correct for JS. Typically use-case: routing through Glacier2.